### ISSUE TITLE
NavigationDrawerDestination items do not follow theme background

### ISSUE THREAD

**Issue by mavriq-dev** [rydmike/flex_color_scheme#219]
The background color of NavigationDrawerDestination items do not follow the theme and the simulator.

This is how the them appears:
![image](https://github.com/rydmike/flex_color_scheme/assets/1278955/9d8ab9eb-62a3-420a-8a42-6ea1aab190dc)

In the simlator it looks like this:
![image](https://github.com/rydmike/flex_color_scheme/assets/1278955/52046d8d-a766-42cf-bfdd-9a35898217d7)

I took the theme code directly from the playground. 

Example Code:

```dart
import 'package:flutter/material.dart';
import 'package:flex_color_scheme/flex_color_scheme.dart';

void main() {
  runApp(const MainApp());
}

class MainApp extends StatelessWidget {
  const MainApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: theme(),
      home: Scaffold(
        appBar: AppBar(),
        drawer: getNavDrawer(),
        body: const Center(
          child: Text('Hello World!'),
        ),
      ),
    );
  }
}

NavigationDrawer getNavDrawer() {
  return const NavigationDrawer(children: [
    NavigationDrawerDestination(icon: Icon(Icons.close), label: Text('Destination 1')),
    NavigationDrawerDestination(icon: Icon(Icons.close), label: Text('Destination 2')),
    NavigationDrawerDestination(icon: Icon(Icons.close), label: Text('Destination 3'))
  ]);
}

// Theme config for FlexColorScheme version 7.3.x. Make sure you use
// same or higher package version, but still same major version. If you
// use a lower package version, some properties may not be supported.
// In that case remove them after copying this theme to your app.
ThemeData theme() => FlexThemeData.light(
      scheme: FlexScheme.materialBaseline,
      surfaceMode: FlexSurfaceMode.levelSurfacesLowScaffold,
      blendLevel: 7,
      subThemesData: const FlexSubThemesData(
        blendOnLevel: 10,
        blendOnColors: false,
        useTextTheme: true,
        useM2StyleDividerInM3: true,
        alignedDropdown: true,
        useInputDecoratorThemeInDialogs: true,
      ),
      visualDensity: FlexColorScheme.comfortablePlatformDensity,
      useMaterial3: true,
      swapLegacyOnMaterial3: true,
      // To use the Playground font, add GoogleFonts package and uncomment
      // fontFamily: GoogleFonts.notoSans().fontFamily,
    );
```

**Comment by rydmike** [comment]
Hi @mavriq-dev,

Thanks for this report. It is indeed a correct observation, it is also sadly expected behavior with current and latest Flutter version 3.19.2.

Let's study it and find out more. 

We can repeat the same issue without `FlexThemeData` with Flutter stable 3.19.2. Btw thanks for an excellent and to the point reproduction sample, that I can now easily change to vanilla Flutter and we get the same result:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MainApp());
}

class MainApp extends StatelessWidget {
  const MainApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: theme(),
      home: Scaffold(
        appBar: AppBar(),
        drawer: getNavDrawer(),
        body: const Center(
          child: Text('Hello World!'),
        ),
      ),
    );
  }
}

NavigationDrawer getNavDrawer() {
  return const NavigationDrawer(children: [
    NavigationDrawerDestination(
        icon: Icon(Icons.close), label: Text('Destination 1')),
    NavigationDrawerDestination(
        icon: Icon(Icons.close), label: Text('Destination 2')),
    NavigationDrawerDestination(
        icon: Icon(Icons.close), label: Text('Destination 3'))
  ]);
}

ColorScheme scheme = ColorScheme.fromSeed(seedColor: Color(0xFF6750A4));

ThemeData theme() => ThemeData(
      colorScheme: scheme,
      navigationDrawerTheme: NavigationDrawerThemeData(
        backgroundColor: scheme.surface,
      ),
    );

```

And it looks the same:

![Screenshot 2024-02-29 at 22 24 00](https://github.com/rydmike/flex_color_scheme/assets/39990307/de577896-09af-4577-9ee9-d09cfb2c9275)

So why does this happen?

It is because the `Drawer` uses `elevation` 1 by default in M3 and the `colorScheme.surface` color that it uses as background on the `Material` it is built from, gets elevation tint based on elevation 1.

However, the `NavigationDrawerDestination` now use the same background color `colorScheme.surface`, but its items are not elevated, so they are using the closer to white version of unelevated `colorScheme.surface`. If you set the `NavigationDrawerThemeData.evelation` to 0, you can see that the colors match.

```dart
ThemeData theme() => ThemeData(
      colorScheme: scheme,
      navigationDrawerTheme: NavigationDrawerThemeData(
        backgroundColor: scheme.surface,
        elevation: 0,
      ),
    );
```

The colors match:

![Screenshot 2024-02-29 at 22 33 11](https://github.com/rydmike/flex_color_scheme/assets/39990307/d1758248-1cf6-4e2b-b2f1-98867710602b)


But what if I want to use e.g. elevation 12? It would then look like this:

![Screenshot 2024-02-29 at 22 34 13](https://github.com/rydmike/flex_color_scheme/assets/39990307/b5a74827-1205-4efe-a3a1-583bf81ae931)

Not pretty and probably not what was expected either!

The simple workaround is to make the `NavigationDrawerDestination` item backgrounds transparent

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MainApp());
}

class MainApp extends StatelessWidget {
  const MainApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: theme(),
      home: Scaffold(
        appBar: AppBar(),
        drawer: getNavDrawer(),
        body: const Center(
          child: Text('Hello World!'),
        ),
      ),
    );
  }
}

NavigationDrawer getNavDrawer() {
  return const NavigationDrawer(children: [
    NavigationDrawerDestination(
        backgroundColor: Colors.transparent,
        icon: Icon(Icons.close),
        label: Text('Destination 1')),
    NavigationDrawerDestination(
        backgroundColor: Colors.transparent,
        icon: Icon(Icons.close),
        label: Text('Destination 2')),
    NavigationDrawerDestination(
        backgroundColor: Colors.transparent,
        icon: Icon(Icons.close),
        label: Text('Destination 3'))
  ]);
}

ColorScheme scheme = ColorScheme.fromSeed(seedColor: const Color(0xFF6750A4));

ThemeData theme() => ThemeData(
      colorScheme: scheme,
      navigationDrawerTheme: NavigationDrawerThemeData(
        backgroundColor: scheme.surface,
        elevation: 12,
      ),
    );

```

![Screenshot 2024-02-29 at 22 36 06](https://github.com/rydmike/flex_color_scheme/assets/39990307/ed70f647-ae0c-4145-9f0f-e65aea1e40f1)

But we are using themes and should not need to think about such things.

Here is a really interesting part. The published **Themes Playground** app does **not do this** with the `NavigationDrawerDestination` items. 

It looks like you showed: 

![Screenshot 2024-02-29 at 22 48 39](https://github.com/rydmike/flex_color_scheme/assets/39990307/176064d1-6fc5-4c6e-9bff-d05d4d96fbfd)

and I double checked, I don't do any transparent color on the destinations in its code.

**Wait, what is going on here?**

Well Themes Playground **live version** was built back with **Flutter SDK 3.13.1**:

![Screenshot 2024-02-29 at 22 50 38](https://github.com/rydmike/flex_color_scheme/assets/39990307/24a33b22-82cf-4157-9c45-bec13778a9b4)

If I build it with latest **Flutter SDK 3.19.2**:

**I get the same result and yes imo issue too!** 

![Screenshot 2024-02-29 at 22 52 17](https://github.com/rydmike/flex_color_scheme/assets/39990307/54a94300-980f-4af5-8815-6fc0ec08add6)

As we already established this is what it looks like now with vanilla Flutter 3.19.2 as well. So it is not a FlexColorScheme issue. 

### What has happened?

Between Flutter version **3.13.1** and **3.19.2**, somewhere along the way there was a change that caused this issue. It perhaps an intentional change, but potentially also a regression.

The change has probably been caused by the color being transparent on the `NavigationDrawerDestination` before, to it now being defined by the Material-3 spec token that has it at surface color.

https://m3.material.io/components/navigation-drawer/specs

Well actually the spec says it should be token `md.sys.color.surface-container-low` as background when it is a modal **Drawer**. Which is a new color that **does not exist** in Flutter yet, but it is equivalent to the elevation level-1 tinted one.

All the elevation tints are going away in Material3 and Flutter too, in favor of precomputed surface shades for different usages. The token `md.sys.color.surface-container-low` is one of those that should be on the modal Drawer and as background on its unselected items.

So if we would have that token color as background color and no elevation tint anymore, the new theme defaults would work and the colors would match, but since we don't, one gets elevation tint (drawer) and the destinations do not, and hence the **colors do not match**.

So yes, I think it is a Flutter bug/regression with too eager implementation of component theme behavior from token db, before Flutter actually support the needed new colors and behavior.

There is PR in Flutter SDK that is preparing the theme for massive changes. I tweeted about it here: https://x.com/RydMike/status/1761057220047909295?s=20

## Conclusion

Sadly since this is a **Flutter SDK** issue and regression, there is not much I can do about it to fix it in FlexColorScheme. I will check if the issue has been reported in Flutter issues and report it as regression, if does not exist. I need to find the stable Flutter version where it broke.

I have not seen any breaking change notification about this, so it should at least be considered a regression.

BTW, thanks for the report, a very good find! 👍🏻 😄 💙

**Comment by rydmike** [comment]
Even if this is "Flutter SDK" sourced and there is not much I can do about it here. I will keep it open so other can discover and learn about it too.

And I will link issue and progress in Flutter repo as well.I will also add it as a known Flutter theming issues on my FlexColorScheme docs site.

**Comment by mavriq-dev** [comment]
Cool. At least I know why it happens and how to work around it. Thanks for the great explanation!

**Comment by rydmike** [comment]
seem like this issue has been solved in Flutter 3.22 already and now with FCS version 8.0.0-dev.1 being out that supports Flutter 3.22 and later, it should no longer be an issue.

---

## INFO

**FlexColorScheme** package dev release 8.0.0-dev.1 has been published. This is a BIG update with breaking changes and many new features. It has its own dev pre-release of the [**Themes Playground**](https://rydmike.com/flexcolorscheme/themesplayground-dev) as well, feel free to take a look. No new API changes are planned for the stable 8.0.0 release from this dev.1 release.

We would appreciate any feedback about found issues from early adopters. 

* [Doc site has some pre-info](https://docs.flexcolorscheme.com/)
* [Dev release of PKG here](https://pub.dev/packages/flex_color_scheme/versions/8.0.0-dev.1)
* [The PKG changelog has the most detailed info](https://pub.dev/packages/flex_color_scheme/versions/8.0.0-dev.1/changelog)

This issue is solved in the new release, but I will not close this issue until the stable version of 8.0.0 has been released. 

**FOR INFO** there will be no new features or changes in it compared to this one. Only any issues found from the quality assurance will be addressed. The stable release will also include many doc updates, but docs site will continue to be edited and revised AFTER the 8.0.0 stable release.

**Comment by rydmike** [comment]
Closing this issue, as it is no longer an issue in the latest release and Flutter version it uses: https://pub.dev/packages/flex_color_scheme
